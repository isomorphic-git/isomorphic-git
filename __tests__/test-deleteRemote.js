/* eslint-env node, browser, jasmine */
const { makeFixture } = require('./__helpers__/FixtureFS.js')
// @ts-ignore
const snapshots = require('./__snapshots__/test-deleteRemote.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')
const { deleteRemote, listRemotes } = require('isomorphic-git')

describe('deleteRemote', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('deleteRemote', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-deleteRemote')
    const remote = 'foo'
    // Test
    await deleteRemote({ dir, gitdir, remote })
    const a = await listRemotes({ dir, gitdir })
    expect(a).toEqual([{ remote: 'bar', url: 'git@github.com:bar/bar.git' }])
  })
  it('missing argument', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-addRemote')
    // Test
    let error = null
    try {
      // @ts-ignore
      await deleteRemote({ dir, gitdir })
    } catch (err) {
      error = err
    }
    expect(error).not.toBeNull()
    expect(error.toJSON()).toMatchSnapshot()
  })
})
