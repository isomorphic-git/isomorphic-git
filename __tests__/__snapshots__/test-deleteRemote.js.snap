// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteRemote missing argument 1`] = `
Object {
  "caller": "git.deleteRemote",
  "code": "MissingRequiredParameterError",
  "data": Object {
    "function": "deleteRemote",
    "parameter": "remote",
  },
  "message": "The function \\"deleteRemote\\" requires a \\"remote\\" parameter but none was provided.",
}
`;
